### ISSUE TITLE
v0.14/v0.15 missing on pypi

### ISSUE THREAD

**Issue by dvzrv** [scheibler/khard#215]
Hi! I'm packaging khard for Arch Linux and would like to build 0.14.0, but currently it's still missing from [pypi.org](https://pypi.org/project/khard/).
Please upload the sdist! Thanks! :)

**Comment by lucc** [comment]
I made the release but the pypi project is still managed by @scheibler. I will discuss it with him.

**Comment by lucc** [comment]
PS: Is there any reason you pull from pypi and not from github?

**Comment by lucc** [comment]
Note to self, let' try to automate this: https://docs.travis-ci.com/user/deployment/pypi/

**Comment by dvzrv** [comment]
Mainly for reproducibility reasons we usually use the [sdist](https://docs.python.org/3.7/distutils/sourcedist.html) release from pypi, unless there is none.

**Comment by DamienCassou** [comment]
I'm using pipsi to get khard and I have the same problem.

**Comment by lucc** [comment]
I contacted @scheibler but until he reacts there is nothing I can do for you, sorry :( I don't have access to the pypi project.

**Comment by doronbehar** [comment]
BTW all of the previous releases up untill 0.10.1 are missing from pypi. This is definitely should be automated.

**Comment by lucc** [comment]
I just released v0.15, sadly @scheibler is not back yet so the problems discussed above are not solved.

For all packagers: This release includes a lot of new documentation including man pages which you might want to ship :)

**Comment by doronbehar** [comment]
Thanks for notifying us never the less :)

**Comment by dvzrv** [comment]
@lucc Thanks, but this release introduces yet another setback, as `setuptools_scm` is now in use:

```
LookupError: setuptools-scm was unable to detect version for '/build/khard/src/khard-0.15.0'.

Make sure you're either building from a fully intact git repository or PyPI tarballs. Most other sources (such as GitHub's tarballs, a git checkout without the .git folder) don't contain the necessary metadata and will not work.

For example, if you're using pip, instead of https://github.com/user/proj/archive/master.zip use git+https://github.com/user/proj.git#egg=proj
```

This will now require me to git clone the tagged version, without any validity check at all.
I'm not comfortable to decrease the security of a package, because this technical issue was not resolved and will therefore remain on `v0.14.0` until it is resolved.

Thanks for the update on the status though! I hope @scheibler is not M.I.A.

**Comment by lucc** [comment]
@dvzrv I don't really know how to solve this. Are there other AUR packages that use setuptools_scm? (I don't have any installed according to `find  ~/.cache/yay -name setup.py |xargs grep setuptools_scm`).

**Comment by lucc** [comment]
@dvzrv maybe #230 can help you.

**Comment by dvzrv** [comment]
@lucc yes, I could patch around this, but given, that @scheibler [is active](https://github.com/scheibler/khard/pull/232#issuecomment-546242643), I don't see any point in making more work for myself, than need be.

**Comment by untitaker** [comment]
@dvzrv use the pypi tarball instead of the github one!

**Comment by lucc** [comment]
@untitaker where is the tarbal for v0.14 or v0.15? I just checked and did not see it. The whole point of this issue that these tarballs are missing.

**Comment by untitaker** [comment]
Sorry, got it.

**Comment by kmosiejczuk** [comment]
> PS: Is there any reason you pull from pypi and not from github?

Just to add another voice to why some folks prefer pypi:

Github has in the past changed how it auto-generates the tarballs it provides. OpenBSD's ports tree keeps hashes of the sources to prevent malicious changes. When github has changed how the tarballs get generated, anything using the auto-generated tarballs breaks. For projects that we have to use github for, we sometimes will file issues to have a static tarball uploaded to the release to avoid these problems.

**Comment by scheibler** [comment]
Done.

0.15.0 contained a small readme format issue. Therefore I've released 0.15.1 (and 0.14.0 for sake of completeness) on pypi.
